spec/components/Groups/Groups.test.jsx

```diff
@@ -0,0 +1,282 @@
+
+        if (currentGroup) {
+          mockedGroups = [currentGroup];
+        } else {
+          mockedGroups = [];
+        }
+      }
+
+      return (
+        <CioPlp
+          apiKey={DEMO_API_KEY}
+          urlHelpers={{ setUrl: setCurrentUrl, getUrl: () => currentUrl }}>
+          <Groups groups={mockedGroups} />
+          <div id='request-filters'>{filters}</div>;
+        </CioPlp>
+      );
+    }
+
+    const getRequestFilters = (container) =>
+      JSON.parse(container.querySelector(`#request-filters`).textContent);
+
+    const currentGroup = { displayName: 'All', groupId: 'all' };
+    const firstLevelGroup = { displayName: 'All Greeting Cards', groupId: '1' };
+    const secondLevelGroup = { displayName: 'Halloween Greeting Cards', groupId: '12' };
+
+    it('Upon clicking a selected option, should navigate to the next level hierarchy', async () => {
+      const { container, getByText } = render(<TestGroups />);
+
+      fireEvent.click(getByText(firstLevelGroup.displayName));
+
+      await waitFor(() => {
+        const requestFilters = getRequestFilters(container);
+        expect(requestFilters.group_id[0]).toBe(firstLevelGroup.groupId);
+        const firstLevelGroupCrumb = container.querySelectorAll('.cio-crumb')[1];
+        const secondLevelGroupEl = container.querySelectorAll('.cio-filter-option-name')[0];
+
+        expect(firstLevelGroupCrumb).not.toBeUndefined();
+        expect(firstLevelGroupCrumb.textContent).toBe(firstLevelGroup.displayName);
+        expect(secondLevelGroupEl).not.toBeUndefined();
+        expect(secondLevelGroupEl.textContent).toBe(secondLevelGroup.displayName);
+      });
+    });
+
+    it('Upon clicking base breadcrumb, should navigate back to base group with no filters', async () => {
+      const { container, getByText } = render(<TestGroups />);
+
+      // Navigate to first level group
+      fireEvent.click(getByText(firstLevelGroup.displayName));
+
+      await waitFor(() => {
+        const baseGroupCrumb = container.querySelectorAll('.cio-crumb')[0];
+
+        expect(baseGroupCrumb.nodeName).toBe('BUTTON');
+
+        if (baseGroupCrumb.nodeName === 'BUTTON') {
+          // Navigate back to base group
+          fireEvent.click(baseGroupCrumb);
+        }
+      });
+
+      await waitFor(() => {
+        const requestFilters = getRequestFilters(container);
+        const breadcrumbs = container.querySelectorAll('.cio-crumb');
+        const firstLevelOptionEl = container.querySelectorAll('.cio-filter-option-name')[0];
+
+        expect(requestFilters.group_id).toBeUndefined();
+        expect(breadcrumbs.length).toBe(1);
+        expect(breadcrumbs[0].textContent).toBe(currentGroup.displayName);
+        expect(firstLevelOptionEl).not.toBeUndefined();
+        expect(firstLevelOptionEl.textContent).toBe(firstLevelGroup.displayName);
+      });
+    });
+
+    it('Upon clicking nested breadcrumb, should navigate back to group with correct groupId filters', async () => {
```

Comment:
You already did the work for whoever is going to implement the breadcrumb component 😅